src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs

```diff
@@ -329,8 +327,33 @@ public void Dispose()
 
         private void DisposeCache()
         {
-            _cached?.Dispose();
-            _cached = null;
+            //If _cached is not null - it means that the client approached with this multitask request to node and the request failed.
+            //and now client tries to send it to another node.
+            if (_cached != null)
+            {
+                var cachedValues = _cached.Values;
+                if (cachedValues != null)
+                {
+                    foreach (var cached in cachedValues)
+                    {
+                        cached.Release.NotModified();
+                    }
+                }
+
+                _cached.Dispose();
+                _cached = null;
+
+                // The client sends the commands.
+                // Some of which could be saved in cache with a response 
+                // that includes the change vector that received from the old fallen node.
+                // The client can't use those responses because their URLs are different 
+                // (include the IP and port of the old node), because of that the client 
+                // needs to get those docs again from the new node.
+                foreach (var command in _commands)
+                {
+                    command.Headers.Remove(Constants.Headers.IfNoneMatch);
```

Comment:
Here you need to update with the `_cached` etags, if it exists.

Reply:
I dispose the `_cached` just after this.
https://github.com/ravendb/ravendb/blob/667d318809805dd6c055f085937833847b654c86/src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs#L334

Reply:
Aren't you going to run this again on a different node? Will we re-get those form the cache, then?

Reply:
If the response was successful, it will be updated in
https://github.com/ravendb/ravendb/blob/667d318809805dd6c055f085937833847b654c86/src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs#L192

Reply:
No, after the node is down it'll re-get those from the node that is still working.
https://github.com/ravendb/ravendb/blob/667d318809805dd6c055f085937833847b654c86/src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs#L193